erasure-coding/benches/README.md

```diff
@@ -0,0 +1,39 @@
+### Run benches
+```
+$ cd erasure-coding # ensure you are in the right directory
+$ cargo bench
+```
+
+### `scaling_with_validators`
+
+This benchmark evaluates the performance of constructing the chunks and the erasure root from PoV and
+reconstructing the PoV from chunks. You can see the results of running this bench on 5950x below.
+Interestingly, with `10_000` chunks (validators) its slower than with `50_000` for both construction
+and reconstruction.
+```
+construct/200           time:   [93.924 ms 94.525 ms 95.214 ms]
+                        thrpt:  [52.513 MiB/s 52.896 MiB/s 53.234 MiB/s]
+construct/500           time:   [111.25 ms 111.52 ms 111.80 ms]
+                        thrpt:  [44.721 MiB/s 44.837 MiB/s 44.946 MiB/s]
+construct/1000          time:   [117.37 ms 118.28 ms 119.21 ms]
+                        thrpt:  [41.941 MiB/s 42.273 MiB/s 42.601 MiB/s]
+construct/2000          time:   [125.05 ms 125.72 ms 126.38 ms]
+                        thrpt:  [39.564 MiB/s 39.772 MiB/s 39.983 MiB/s]
+construct/10000         time:   [270.46 ms 275.11 ms 279.81 ms]
+                        thrpt:  [17.869 MiB/s 18.174 MiB/s 18.487 MiB/s]
```

Comment:
This is quite interesting. Would be interesting to see at which number of validators the tput reverses trend ?

Reply:
The AFFT scales like `blocksize * log(validators)` but with the log being discrete.

Reply:
We'll collapse from all those `validators^2` gossip messages long before this poses any issues.  We just need enough validators per relay chain to reach 2nd layer scaling, which is only 600 by Bryan Ford's estimates in OmniLedger, but is more like 1000 by Alfonso's estimates.  We'd go to 3/4 of a power of 2 to optimize the erasure coding if we can really make 769 work or whatever, but maybe 1500 is a bit large for the gossip.

Reply:
> It's arithmetic cannot handle more than 16k validators, so we're missing asserts that should kill it long before 50k.

Are you sure it's not 2^16, which is 64k? I added an assertion in reconstruct that it decodes to the original PoV and it passes with 50k validators, but panics with `TooManyValidators` error with 70k shards.

Reply:
> We'll collapse from all those validators^2 gossip messages long before this poses any issues. 

I agree, at this point (10k+ validators) networking would be the bottleneck, not CPU cost of erasure-coding.

Reply:
oops ye you're right, 2^16.  lol

Reply:
It's a log factor so I guess some unrelated artifact of the benchmark is just overriding this somehow.  the erasure coding cannot actually get faster with more validators